auth/oauth/types.go

```diff
@@ -397,3 +400,46 @@ type OpenIDCredentialIssuerMetadata struct {
 	// - Display: a slice of maps where each map represents the display information (optional)
 	Display []map[string]string `json:"display,omitempty"`
 }
+
+// OpenIDConfiguration represents the OpenID configuration
+// It contains the minimal information required for OpenID4VP, the required `jwks` is also omitted
+// see https://openid.net/specs/openid-connect-federation-1_0-29.html#entity-statement
+type OpenIDConfiguration struct {
+	// - Issuer: an url representing the issuer of the entity statement
+	// for now we keep it teh same as the subject, eg the subject/tenant
+	Issuer string `json:"iss"`
+	// - Subject: an url representing the subject of the entity statement
+	Subject string `json:"sub"`
+	// IssuedAt: the time the entity statement was issued
+	IssuedAt int64 `json:"iat"`
+	// JWKs is the JSON Web Key Set of the entity statement. Contains keys of all DIDs for the subject
+	JWKs jwk.Set `json:"jwks"`
+	// OpenIDProvider: the metadata of the OpenID provider. We need it as Authorization server metadata
+	OpenIDProvider AuthorizationServerMetadata `json:"openid_provider"`
```

Comment:
shouldn't this be wrapped in a `metadata` field? (according to the example in https://openid.net/specs/openid-connect-federation-1_0-29.html#entity-statement)